src/main/java/com/waytta/SaltAPIBuilder.java

```diff
@@ -135,32 +129,48 @@ public void setKwarguments(String kwarguments) {
         this.kwarguments = kwarguments;
     }
 
-    public String getClientInterface() {
-        return clientInterfaceName;
-    }
-
     public Boolean getBlockbuild() {
-        return clientInterface.getBlockBuild();
+        if (clientInterface instanceof LocalClient) {
+        	return ((LocalClient) clientInterface).getBlockbuild();
+        } else {
+        	return false;
+        }
     }
 
     public String getBatchSize() {
-        return clientInterface.getBatchSize();
+        if (clientInterface instanceof LocalBatchClient) {
+        	return ((LocalBatchClient) clientInterface).getBatchSize();
+        } else {
+        	return "";
+        }
     }
 
     public Integer getJobPollTime() {
-        return clientInterface.getJobPollTime();
+        if (clientInterface instanceof LocalClient) {
+        	return ((LocalClient) clientInterface).getJobPollTime();
+        } else {
+        	return 0;
+        }
     }
     
     public String getMods() {
-    	return clientInterface.getMods();
+        if (clientInterface instanceof RunnerClient) {
+        	return ((RunnerClient) clientInterface).getMods();
+        } else {
+        	return "";
+        }
     }
     
     public String getPillarvalue() {
-    	return clientInterface.getPillarValue();
+        if (clientInterface instanceof RunnerClient) {
+        	return ((RunnerClient) clientInterface).getPillarvalue();
+        } else {
+        	return "";
```

Comment:
Seems like a failure of design; rather define an `abstract` method in `BasicClient`.